routing/index_graph.hpp

```diff
@@ -77,16 +187,169 @@ class IndexGraph final
   {
     uint32_t const jointsSize = ReadPrimitiveFromSource<uint32_t>(src);
     m_roadIndex.Deserialize(src);
-    m_jointIndex.Build(m_roadIndex, jointsSize);
+    Build(jointsSize);
+  }
+
+  Joint::Id GetJointIdForTesting(RoadPoint const & rp) const
+  {
+    return m_roadIndex.GetJointId(rp);
+  }
+
+  /// \brief Disable an edges between |from| and |to| along |featureId|.
+  /// \note Despite the fact that |from| and |to| could be connected with several edges
+  /// it's a rare case. In most cases |from| and |to| are connected with only one edge
+  /// if they are adjacent.
+  /// \note The method doesn't affect routing if |from| and |to| are not adjacent or
+  /// if one of them is equal to Joint::kInvalidId.
+  void DisableEdge(DirectedEdge const & edge)
+  {
+    m_blockedEdges.insert(edge);
+  }
+
+  void DisableAllEdges(Joint::Id from, Joint::Id to)
+  {
+    vector<pair<RoadPoint, RoadPoint>> result;
+    m_jointIndex.FindPointsWithCommonFeature(from, to, result);
+    for (auto const & p : result)
+      DisableEdge(DirectedEdge(from, to, p.first.GetFeatureId()));
+  }
+
+  /// \brief Adding a fake oneway feature with a loose end starting from joint |from|.
+  /// Geometry for the feature points is taken from |geometrySource|.
+  /// If |geometrySource| contains more than two points the feature is created
+  /// with intermediate (not joint) point(s).
+  /// \returns feature id which was added.
+  uint32_t AddFakeLooseEndFeature(Joint::Id from, vector<RoadPoint> const & geometrySource,
+                                  double speed);
+
+  /// \brief Connects joint |from| and |to| with a fake oneway feature. Geometry for the feature
+  /// points is taken from |geometrySource|. If |geometrySource| contains more than
+  /// two points the feature is created with intermediate (not joint) point(s).
+  /// \returns feature id which was added.
+  uint32_t AddFakeFeature(Joint::Id from, Joint::Id to, vector<RoadPoint> const & geometrySource,
```

Comment:
I would prefer to consider it as adding a fake feature. 
AddFakeFeature() always adds an edge.
But AddFakeLooseEndFeature() does not add an edge. It's a feature. So for uniformity I suggest to consider them as fake features which could be edges.